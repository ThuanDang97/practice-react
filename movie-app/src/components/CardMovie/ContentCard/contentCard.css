.content h2 {
  font-size: 1em;
  margin: 0;
  width: 100%;
  word-wrap: normal;
  overflow-wrap: break-word;
}

.content h2 a {
  font-weight: 700;
  color: #000;
}

.content h2 a:hover {
  font-weight: 700;
  color: rgb(1, 180, 228);
}

.content p {
  font-size: 1em;
  margin: 0;
  padding: 0;
  color: rgba(0, 0, 0, 0.6);
}

.consensus {
  position: absolute;
  top: -19px;
  left: 10px;
  display: inline-block;
  transition: transform 0.2s;
  transform: scale(1);
}

.tight {
  width: 38px;
  height: 38px;
  box-sizing: border-box;
}

.outer_ring {
  margin-right: 0;
  width: 38px;
  height: 38px;
  padding: 2px;
  display: inline-block;
  border-radius: 50%;
  background-color: #081c22;
}

.user_score_chart {
  position: relative;
  display: inline-block;
  width: 100%;
  height: 100%;
  text-align: center;
}

.percent {
  width: 100%;
  height: 100%;
  z-index: 2;
  display: flex;
  align-items: center;
  justify-content: center;
}

.icon {
  padding-top: 1px;
  padding-left: 1px;
  font-size: 20px;
  color: #fff;
}
