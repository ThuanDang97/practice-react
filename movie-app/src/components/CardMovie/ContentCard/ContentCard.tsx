import { memo } from 'react';
import { Link } from 'react-router-dom';
import { convertRating } from '../../../utils';
import './contentCard.css';

type TProps = {
  id: number;
  title: string;
  voteAverage: number;
  releaseDate: string;
};

const ContentCard = (props: TProps): JSX.Element => {
  const { id, title, voteAverage, releaseDate } = props;

  const point = convertRating(voteAverage);

  return (
    <div className="content">
      <div className="consensus tight">
        <div className="outer_ring">
          <div
            className="user_score_chart"
            data-percent="78.0"
            data-track-color="#204529"
            data-bar-color="#21d07a"
          >
            <div className="percent">
              <span className="icon">{point}</span>
            </div>
          </div>
        </div>
      </div>
      <h2>
        <Link to={`/movie/${id}`} title={title}>
          {title}
        </Link>
      </h2>
      <p>{releaseDate}</p>
    </div>
  );
};

export default memo(ContentCard);
