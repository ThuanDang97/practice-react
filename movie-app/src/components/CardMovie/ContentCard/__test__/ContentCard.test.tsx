import { screen, render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import ContentCard, { TProps } from '../ContentCard';

describe('Test Content Card component', () => {
  const defaultProps: TProps = {
    id: 1,
    releaseDate: '2022-05-04',
    title: 'Doctor Strange in the Multiverse of Madness',
    voteAverage: 7.5,
  };

  const component = (
    <Router>
      <ContentCard {...defaultProps} />;
    </Router>
  );

  test('Component Content Card Movie matches DOM Snapshot', () => {
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Movie should render correctly', () => {
    render(component);

    expect(screen.getByTestId('score')).toHaveTextContent('75');
    expect(screen.getByRole('heading')).toHaveTextContent(
      'Doctor Strange in the Multiverse of Madness',
    );
    expect(screen.getByTestId('release-date')).toHaveTextContent('2022-05-04');
  });
});
