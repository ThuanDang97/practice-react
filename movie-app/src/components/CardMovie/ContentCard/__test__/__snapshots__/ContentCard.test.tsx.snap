// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Content Card component Component Content Card Movie matches DOM Snapshot 1`] = `
<div>
  <div
    class="content"
  >
    <div
      class="consensus tight"
    >
      <div
        class="outer_ring"
      >
        <div
          class="user_score_chart"
          data-bar-color="#21d07a"
          data-percent="78.0"
          data-track-color="#204529"
        >
          <div
            class="percent"
          >
            <span
              class="icon"
              data-testid="score"
            >
              75
            </span>
          </div>
        </div>
      </div>
    </div>
    <h2>
      <a
        href="/movie/1"
        title="Doctor Strange in the Multiverse of Madness"
      >
        Doctor Strange in the Multiverse of Madness
      </a>
    </h2>
    <p
      data-testid="release-date"
    >
      2022-05-04
    </p>
  </div>
  ;
</div>
`;
